scripts/us_census/pep/us_pep_sexrace/postprocess.py

```diff
@@ -0,0 +1,370 @@
+                df4[col] = df4[col].astype("str")
+            df5 = pd.concat([df4, df5], ignore_index=True)
+
+        # coverting year values to int
+        df5['Year'] = df5['Year'].astype(float).astype(int)
+
+        # making geoid uniform
+        df5['geo_ID'] = df5['geo_ID'].str.strip()
+
+        # sorting the values based on year and geoid
+        df5.sort_values(by=['Year', 'geo_ID'], ascending=True, inplace=True)
+
+        # dropping unwanted column
+        df5 = df5.drop(columns=['Unnamed: 0','Count_Person_Male',\
+            'Count_Person_Female'])
+
+        # writing output to final csv
+        df5.to_csv(_CODEDIR + "/output_files/final/" +
+                   "sex_race_aggregate_national_after.csv",
+                   index=False)
+
+        # collecting all the column headers
+        columns_of_geo_aggregate_output_files = df5.columns.to_list()
+
+        column_names[4] = columns_of_geo_aggregate_output_files
+    return column_names
+
+
+def generate_mcf(statvar_names: list, flag1: int) -> None:
+    """
+    This method generates 3 MCF file w.r.t
+    dataframe headers and defined MCF template
+    1. sex_race.mcf : for the files which are processed as is.
+    2. sex_race_aggregate.mcf : for the files which
+        are having Count Person Male/Female aggregated.
+    3. sex_race_aggregate_state_2010_2020.mcf : for the files
+        which are aggregated from different geo granularity.
+
+    Args:
+        sv_names (list) : List of DataFrame Columns
+        flag1 (int) : flag value helps in generating
+        output files. Possible values are 1,2 and 3.
+        1 - generate mcf for files which are processed as is
+        2 - generate mcf for files which are having
+        Count_Person_Male and Count_Person_Female aggregated
+        3 - generate mcf for files which are aggregated
+        from different geo granulartiy.
+
+    Returns:
+        None
+    """
+    mcf_template = """Node: dcid:{}
+typeOf: dcs:StatisticalVariable
+populationType: dcs:Person
+{}{}
+statType: dcs:measuredValue
+measuredProperty: dcs:count
+"""
+
+    final_mcf_template = ""
+    for sv in statvar_names:
+        if "Total" in sv:
+            continue
+        if "Year" in sv:
+            continue
+        if "geo_ID" in sv:
+            continue
+        gender = ''
+        race = ''
+        sv_prop = sv.split("_")
+        for prop in sv_prop:
+            if prop in ["Count", "Person"]:
+                continue
+            if "Male" in prop or "Female" in prop:
+                gender = "gender: dcs:" + prop
+            else:
+                race = "race: dcs:" + prop + "\n"
+        final_mcf_template += mcf_template.format(sv, race, gender) + "\n"
+    # Writing Genereated MCF to local path.
+    if flag1 == 1:
+        with open(_CODEDIR + "/output_files/final/" + "sex_race.mcf",
+                  'w+',
+                  encoding='utf-8') as f_out:
+            f_out.write(final_mcf_template.rstrip('\n'))
```

Comment:
lines 261-264 are repeated a few times, could we abstract them out as a helper function?

Reply:
This helper function can also be re-used in generate_tmcf().

Reply:
Abstracted the same lines of code to helper function for generate_mcf and generate_tmcf.